src/test/java/com/microsoft/azure/relay/AsyncLockTest.java

```diff
@@ -206,4 +206,28 @@ public void lockScopeUncontendedThrow() throws Throwable {
 		assertEquals(IOException.class, error.getCause().getClass());
 		assertFalse(lock.isLocked());
 	}
+	
+	@Test
+	public void lockScopeContendedTimeout() throws Throwable {
+		AsyncLock lock = new AsyncLock(EXECUTOR);	
+		LockRelease lockRelease = lock.acquireAsync().join();
+		
+		CompletableFuture<Void> innerTaskStart = new CompletableFuture<Void>();
+		CompletableFuture<String> outerTask = lock.lockThenCompose(Duration.ofMillis(TIMEOUT_MS), () -> {
+			assertTrue("Should be locked during Inner Function", lock.isLocked());
+			innerTaskStart.complete(null);			
+			throw new CompletionException(new IOException("Test Induced Exception"));
```

Comment:
>throw new CompletionException(new IOException("Test Induced Exception")); [](start = 3, length = 73)

This line never runs, not truly needed.

Reply:
Does anything in here run? If not, we can probably omit all of it except maybe assert the inside isn't reached.

---
In reply to: [259493522](https://github.com/Azure/azure-relay-java/pull/22#discussion_r259493522) [](ancestors = 259493522)